Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp

```diff
@@ -0,0 +1,351 @@
+    {
+        Name Handler = Name::FromStringLiteral("Handler");
+        Name Label = Name::FromStringLiteral("Label");
+        Name SerializedPath = Name::FromStringLiteral("SerializedPath");
+    } // namespace DescriptorAttributes
+
+    namespace LegacyReflectionInternal
+    {
+        struct InstanceVisitor
+            : IObjectAccess
+            , IAttributes
+        {
+            IReadWrite* m_visitor;
+            SerializeContext* m_serializeContext;
+
+            struct AttributeData
+            {
+                // Group from the attribute metadata (generally empty with Serialize/EditContext data)
+                Name m_group;
+                // Name of the attribute
+                Name m_name;
+                // DOM value of the attribute - currently only primitive attributes are supported,
+                // but other types may later be supported via opaque values
+                Dom::Value m_value;
+            };
+
+            struct StackEntry
+            {
+                void* m_instance;
+                AZ::TypeId m_typeId;
+                const SerializeContext::ClassData* m_classData = nullptr;
+                const SerializeContext::ClassElement* m_classElement = nullptr;
+                AZStd::vector<AttributeData> m_cachedAttributes;
+                AZStd::string m_path;
+            };
+            AZStd::stack<StackEntry> m_stack;
+
+            using HandlerCallback = AZStd::function<bool()>;
+            AZStd::unordered_map<AZ::TypeId, HandlerCallback> m_handlers;
+
+            virtual ~InstanceVisitor() = default;
+
+            InstanceVisitor(IReadWrite* visitor, void* instance, const AZ::TypeId& typeId, SerializeContext* serializeContext)
+                : m_visitor(visitor)
+                , m_serializeContext(serializeContext)
+            {
+                m_stack.push({ instance, typeId });
+                RegisterPrimitiveHandlers<bool, uint8_t, uint16_t, uint32_t, uint64_t, int8_t, int16_t, int32_t, int64_t, float, double>();
+            }
+
+            template<typename T>
+            void RegisterHandler(AZStd::function<bool(T&)> handler)
+            {
+                m_handlers[azrtti_typeid<T>()] = [this, handler]() -> bool
+                {
+                    return handler(*reinterpret_cast<T*>(m_stack.top().m_instance));
+                };
+            }
+
+            template<typename T>
+            void RegisterPrimitiveHandlers()
+            {
+                RegisterHandler<T>(
+                    [this](T& value) -> bool
+                    {
+                        m_visitor->Visit(value, *this);
+                        return false;
+                    });
+            }
+
+            template<typename T1, typename T2, typename... Rest>
+            void RegisterPrimitiveHandlers()
+            {
+                RegisterPrimitiveHandlers<T1>();
+                RegisterPrimitiveHandlers<T2, Rest...>();
+            }
```

Comment:
You don't need to this anymore.
You can use a [fold expression](https://en.cppreference.com/w/cpp/language/fold) to expand the parameter pack in order.

```suggestion
            template<typename... T>
            void RegisterPrimitiveHandlers()
            {
                (RegisterHandler<T>(
                    [this](T& value) -> bool
                    {
                        m_visitor->Visit(value, *this);
                        return false;
                    }), ...);
            }
```

Reply:
Neat, done.